### ISSUE TITLE
Assemblyline Process: recipe have non-unique circuits

### ISSUE THREAD

**Issue by slprime** [GTNewHorizons/GT-New-Horizons-Modpack#16839]
### Your GTNH Discord Username

SLPrime

### Your Pack Version

nightly 550

### Your Server

SP

### Java Version

Java 17

### Type of Server

None

### Your Expectation

I expect to see a unique set of items.

### The Reality

Have: ![image](https://github.com/user-attachments/assets/c86eb469-7eba-4144-b382-5dfe3b9707a7)

### Your Proposal

PositionedStack have been unique. At the moment I found a similar problem only In Assemblyline Process

P.S. (items in tooltip is new function which I'm working on)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by chochem** [comment]
i am not sure I understand. is this a ticket about your own wip feature?

**Comment by slprime** [comment]
> i am not sure I understand. is this a ticket about your own wip feature?

no, I found recipes that have duplicates items in PositionedStack.items. this one breaks shift+scroll, preventing you from scrolling through the entire list

**Comment by slprime** [comment]
We need to correct the recipes for the Assemblyline Process so that the microcircuits do not have duplicates

**Comment by slprime** [comment]
I haven’t worked on GT and recipes before, and I can’t fix it myself

**Comment by Ya9yu** [comment]
> i am not sure I understand. is this a ticket about your own wip feature?

I can confirm that when we tested a new feature with a list in the tooltip, SLPrime discovered a problem due to which the shift + scroll in the assembly line recipe was broken

**Comment by chochem** [comment]
hmm well scrolling seems to work just fine for me (without this feature). The recipes in GT also look fine.

**Comment by slprime** [comment]
in what version of NEI/GTNH you tested?

**Comment by chochem** [comment]
nightly 562 with NotEnoughItems-2.6.18-GTNH-pre

**Comment by dimazha** [comment]
> seems to work just fine

In recipes with ZPM circuits I can roll quantum ZPM circuit. I tried to roll in both directions
![image](https://github.com/user-attachments/assets/036cd413-8cba-4c4b-8a57-afdf239805ea)
![image](https://github.com/user-attachments/assets/8299836b-9914-4fc3-93f2-9f7be7338943)
![image](https://github.com/user-attachments/assets/b25cd1a6-ebee-4381-b48b-831e6e29a099)
![image](https://github.com/user-attachments/assets/8bde1567-51e1-464a-a9bd-faf885680fb9)

**Comment by chochem** [comment]
hmm, there is indeed some weird stuff with the order, but I do get all myself:

![image](https://github.com/user-attachments/assets/705c8271-b9d8-4df2-bf57-5c0236c23d4c)
![image](https://github.com/user-attachments/assets/fe570c8a-6f04-4647-ac59-62629495ad2f)
![image](https://github.com/user-attachments/assets/b511f6a8-68e4-4bec-ab93-78a0debb2aa1)
![image](https://github.com/user-attachments/assets/e660059b-59a9-4207-9336-06b80a763bbb)
![image](https://github.com/user-attachments/assets/1f82dece-ca85-423e-a0e4-02c60af7d288)
![image](https://github.com/user-attachments/assets/80030832-3ca6-4a3d-b497-c28dc0846dd2)

curious.

**Comment by slprime** [comment]
in 562 + gregtech-5.09.48.102.jar

https://github.com/user-attachments/assets/d8a405a2-4aaf-490a-8301-6e9dd9c64f4c

maybe it depends on how the modpack was loaded

**Comment by chochem** [comment]
I see. the recipes themselves are fine, so maybe something in the recipemap handling or the specific NEI handling. AL has some special code for recipe map handling with some extra code for handling oredicts and alternative items. Probably something in there.

Since the scrolling part is inconsistent I guess its easier to debug once your feature is a thing.

**Comment by miaowwwwww** [comment]
a little addition to the pattern editing situation:
when i scroll, i usually don't get 1 or 2 circuit to show, so I scroll very fast(not 1 by 1) and let it stop on some random circuit, this way it is possible to roll the missing circuit.
